dao.py

```diff
@@ -645,6 +665,17 @@ def check_creds_and_get_session_id_or_none(self, username, password):
         else:
             return None
 
+    def check_creds(self, username, password):
```

Comment:
i'm confused, it looks there are now 3 functions which repeat a lot of code?:

1. `check_creds(username, password)` which returns whether (username, password) are valid credentials.
2. `check_creds_and_get_session_id_or_none(username, password)`, which checks if (username, password) are valid credentials, and tries to return the current session
3. `get_and_verify_user_by_username`, which seems to be a duplicate of `get_user_by_username`.

Right now 1 and 2 duplicate a lot of the same code. I recommend removing `get_and_verify_user_by_username` and refactoring 1 and 2 so that check_creds returns a user object (not just a bool), and so that get_session_id calls check_creds to get the user object.